fixed + or - after factorial being ignored

mtettmar opened the pull request:
This fixes the following problem:

5*3!+2 evaluate to 30, but should be 32.  The +2 is being ignored.

Currently we have to use` (5*3!)+2 but 5*3!+2 `should be accepted etc.

This pull requests fixes that.

aviaryan commented on BigEval.js:
```diff
@@ -570,6 +570,12 @@ BigEval.prototype.compile = function (expression) {
 
+			console.log(tokens[i - 2]);
+
+			if (i > 1 && tokens[i - 2].value==='!') {				
```
If i = 2, wouldn't `!` being at token 0 be an invalid case?

I haven't looked at the code in years, so I apologize if I am wrong.

Also, can you please remove the console.log? Thanks.

mtettmar replied:
Oops. Sorry. That should be sorted now.

mtettmar replied:
Re I=2 trouble is ! can also be used as Not operator .. ?

aviaryan replied:
You are right. I forgot we had NOT operator.

aviaryan commented on BigEval.js:
```diff
@@ -570,6 +570,12 @@ BigEval.prototype.compile = function (expression) {
+			if (i > 1 && tokens[i - 2].value==='!') {				
+				continue
+			}
+
```
Adding a test for this fix would be nice too. We have automated tests if you pull the latest commits from `master`.

Can you pull the latest commits from `master` and add a test or two? Thanks.

mtettmar replied:
Have added a test.

mtettmar commented:
Just fixed another issue. This was causing an unhandled execution, thus resulting in 'ERROR':

			if (!left && op === '-') {
				left = [{ type: TokenType.NUMBER, value: 0, source: expression }];
			}

expression does not exist. Removed source: expression to fix.

E.g.:

0.125^(-2/3) - gave 'ERROR' instead of 3.99999...

aviaryan commented:
@mtettmar Merging #25 added conflicts to this MR. Can you please look into them?

mtettmar commented:
Yep no problem - have done that.

mtettmar commented:
Closing - now fixed in #26
